extension/observer/observer.go

```diff
@@ -26,80 +26,40 @@ const (
 	ProtocolTCP Protocol = "TCP"
 	// ProtocolUDP is the UDP protocol.
 	ProtocolUDP Protocol = "UDP"
+	// ProtocolUnknown is some other protocol or it is unknown.
+	ProtocolUnknown Protocol = "Unknown"
 )
 
 // Endpoint is a service that can be contacted remotely.
-type Endpoint interface {
+type Endpoint struct {
 	// ID uniquely identifies this endpoint.
-	ID() string
+	ID string
 	// Target is an IP address or hostname of the endpoint.
-	Target() string
-	// String pretty formats the endpoint.
-	String() string
-	// Labels is a map of arbitrary metadata.
-	Labels() map[string]string
+	Target string
+	// Details contains additional context about the endpoint such as a Pod or Port.
+	Details interface{}
```

Comment:
I know you mentioned this in the PR but just to double check is there any trait that `Details` must exhibit that we can define an interface on? Perhaps it has to have a Name?

Reply:
I think in the future it might have something like `ToMap() map[string]string` that returns a map of the object using semantic conventions but it's not needed yet. If something like Name would be common across all Details then it would go into the Endpoint instead I think. I considered `Type()` but that's really just duplicating the Go type so not much help.